doc/README/Effect/Monad/Partiality.agda

```diff
@@ -0,0 +1,27 @@
+-----------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Example showing the use of the partiality Monad
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe --guardedness #-}
+
+module README.Effect.Monad.Partiality where
```

Comment:
Should we move this to `Data.Nat.McCarthy` instead? We could prove the equivalence to [the tail recursive version](https://en.wikipedia.org/wiki/McCarthy_91_function#History) for instance.

Reply:
That would work too - I have no strong opinion either way.

Reply:
I guess I'd vote against. I don't think the f91 is really a piece of functionality that is going to be useful to anyone. Indeed if they are proving properties about it, it'll almost certainly for pedagogical purposes in which case they'll want to redefine it. Therefore the README seems like a good place for it.